### PR TITLE
Show all money corp receives in preview, incl. divs (fixes #1367)

### LINKED ISSUE #1367: 'Payout' preview should show money paid to bank shares
The payout preview is great, but I think that the 'Treasury' value for 'payout' should show the treasury gain for dividends paid to bank shares.

### PR DESCRIPTION
**Pull request by dfannius**
Also I finally changed "Payout" to "Pay Out"; "payout" is a noun and the button names should be verbs.

### CODE REVIEW

**Comment by dfannius** [comment]
Forgot to run rake, I'm rusty!

**Comment by dfannius** [comment]
By the way, I was originally going to show the money that a corporation gets from dividends in a separate column, but we're already really low on horizontal space, so I just lumped it in together for the Treasury column. The information is published separately, though, so we could break it out in the display if we wanted.

File: lib/engine/step/dividend.rb [hunk 1]
```diff
@@ -88,13 +89,31 @@ def payout(entity, revenue)
         { company: 0, per_share: payout_per_share(entity, revenue)[0] }
       end
 
+      def dividends_for_entity(entity, holder, per_share)
+        return 0 if (percent = holder.percent_of(entity)).zero?
```
Comment (tobymao) [inline lib/engine/step/dividend.rb:93]: can you just do num_shares_of

Comment (dfannius) [reply lib/engine/step/dividend.rb:93]: Sure, although looking at the code I see that `num_shares_of` just calls `percent_of` and divides by 10.

Comment (dfannius) [reply lib/engine/step/dividend.rb:93]: Actually, when I changed `percent_of` to `num_shares_of` a ton of unit tests failed. I can look into why tomorrow.

Comment (dfannius) [reply lib/engine/step/dividend.rb:93]: Never mind, I see what your point was.

File: lib/engine/step/dividend.rb [hunk 2]
```diff
@@ -88,13 +89,29 @@ def payout(entity, revenue)
         { company: 0, per_share: payout_per_share(entity, revenue)[0] }
       end
 
+      def dividends_for_entity(entity, holder, per_share)
+        return 0 if (shares = holder.num_shares_of(entity)).zero?
```
Comment (tobymao) [inline lib/engine/step/dividend.rb:93]: can't this just be holder.num_shares_of(entity) * per_share, why do you need to check for zero

Comment (dfannius) [reply lib/engine/step/dividend.rb:93]: Sure, I was just changing the existing code as little as possible, but it used to have more of a reason to early-out.

**Pull request merged**